### ISSUE TITLE
catch with method groups.

### ISSUE THREAD

**Issue by MkazemAkhgary** [dotnet/csharplang#1069]
Allow to single linear catches with method groups.

```
try 
{
}
catch HandleFoo;
catch when (x) HandleBar;
catch HandleException;

// methods
void HandleFoo(Foo exception)...
void HandleBar(Bar exception)...
void HandleException(Exception exception)...
```

signature of HandleException must have single Exception parameter. it must not be overloaded with another Exception parameter either, else it will be compiler error.

this is not conflicting with general catch statement. since the syntaxes are different.

above code translates to

```
try 
{
}
catch(Foo f)
{
    HandleFoo(f);
}
catch(Bar r) when (x)
{
    HandleBar(r);
}
catch(Exception e)
{
    HandleException(e);
}
```

---

<s>continuation of this proposal requires supporting #1068 

method groups can have return value too.

```
Foo result = try GetFoo(); catch HandleFoo;

// methods
Foo HandleFoo(Exception exception)...
```
</s>

**Comment by stuartstein777** [comment]
So many proposals about making try / catch more concise, which I would love. i hate the verbosity and ugliness of try / catch.

Hopefully one of these proposals eventually works and sticks.

**Comment by MkazemAkhgary** [comment]
@stuartstein777 well at some point `when` clause has made its path there somehow 😺

**Comment by aluanhaddad** [comment]
@stuartstein777 true, but most of then so far would likely lead to poor practices and obscure code. I don't see that as an issue with this proposal at all.

**Comment by gulshan** [comment]
Have you seen my closed issue- #831 ?

**Comment by aluanhaddad** [comment]
@gulshan I forgot about that, I would say this is largely a duplicate.
